components/count-badge/count-badge.js

```diff
@@ -172,19 +293,37 @@ class CountBadge extends RtlMixin(LitElement) {
 		if (this.maxDigits && this.number.toString().length > this.maxDigits) {
 			numberString = `${'9'.repeat(this.maxDigits)}+`;
 		}
+		const iconSizeStyle = {};
+		if (this.icon && this._fixIconSizeTier1) {
+			iconSizeStyle.height = '18px';
+			iconSizeStyle.width = '18px';
+		}
 		return html`
-        	<div
-			class="d2l-count-badge-number" 
-			id="${this._badgeId}"
+			<div 
+			class="d2l-count-badge-wrapper"
+			@blur="${this._onBlur}"
+			@focus="${this._onFocus}"
+			id="${ifDefined(this.icon ? undefined : this._badgeId)}"
 			tabindex="${ifDefined(this.tabStop || this.hasTooltip ? '0' : undefined)}" 
 			aria-labelledby="${ifDefined(this.hasTooltip ? undefined : this._textId)}">
-					<div aria-hidden="true">${numberString}</div>
-					${this.hasTooltip ?
-		html`<d2l-tooltip id="${this._textId}" for="${this._badgeId}" aria-live="${this.announceChanges ? 'polite' : 'off'}">${this.text}</d2l-tooltip>`
-		: html`<span id="${this._textId}" class="d2l-offscreen" aria-live="${this.announceChanges ? 'polite' : 'off'}">"${this.text}"</span>`}
-			</div>
+				<div class="d2l-count-badge-number">
+						<div aria-hidden="true">${numberString}</div>		
+				</div>
+				${this.icon ? html`<d2l-icon role="presentation" id="${this._badgeId}" icon="${ifDefined(this.icon)}" class="d2l-button-icon" style=${styleMap(iconSizeStyle)}></d2l-icon>` : null }
```

Comment:
Do you need the `ifDefined` check on the `icon` attribute since if it's `undefined` it would already be rendering `null`.

`role="presentation"` also shouldn't be needed -- if it is then we should be doing something internally in `<d2l-icon>` for that.